wsmaster/che-core-api-logger/src/main/java/org/eclipse/che/api/logger/LoggerService.java

```diff
@@ -0,0 +1,136 @@
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.rest.Service;
+import org.eclipse.che.api.logger.shared.dto.LoggerDto;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Defines Logger REST API. It allows to manage the loggers (with log level) dynamically.
+ *
+ * @author Florent Benoit
+ */
+@Api(value = "/logger", description = "Logger REST API")
+@Path("/logger")
+public class LoggerService extends Service {
+
+  @GET
+  @Path("/{name}")
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(value = "Get the logger level for the given logger")
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The response contains requested logger entity"),
+    @ApiResponse(code = 404, message = "The logger with specified name does not exist")
+  })
+  public LoggerDto getLoggerByName(@ApiParam(value = "logger name") @PathParam("name") String name)
+      throws NotFoundException {
+    return asDto(getLogger(name));
+  }
+
+  @GET
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(
+    value = "Get loggers which are configured",
+    notes = "This operation can be performed only by authorized user",
+    response = LoggerDto.class,
+    responseContainer = "List"
+  )
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The loggers successfully fetched"),
+  })
+  public List<LoggerDto> getLoggers() {
+    LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+
+    return loggerContext
+        .getLoggerList()
+        .stream()
+        .filter(log -> log.getLevel() != null || log.iteratorForAppenders().hasNext())
+        .map(this::asDto)
+        .collect(Collectors.toList());
+  }
+
+  protected LoggerDto asDto(final Logger log) {
+    return newDto(LoggerDto.class).withName(log.getName()).withLevel(log.getLevel().levelStr);
+  }
+
+  @PUT
+  @Path("/{name}")
+  @Consumes(APPLICATION_JSON)
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(value = "Update the logger level")
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The logger successfully updated"),
+  })
+  public LoggerDto updateLogger(
+      @ApiParam(value = "logger name") @PathParam("name") String name, LoggerDto update)
+      throws NotFoundException {
+    Logger logger = getLogger(name);
+    logger.setLevel(Level.toLevel(update.getLevel()));
+    return asDto(logger);
+  }
+
+  @POST
+  @Path("/{name}")
+  @Consumes(APPLICATION_JSON)
+  @Produces(APPLICATION_JSON)
+  @ApiOperation(value = "Create a new logger level")
+  @ApiResponses({
+    @ApiResponse(code = 200, message = "The logger successfully created"),
+  })
+  public LoggerDto createLogger(
```

Comment:
I think this method should be covered with `System#MANAGE_SYSTEM` permission actions. Or maybe even all methods of this service. WDYT?

Reply:
permission mechanism added